### ISSUE TITLE
OpenMP not used after compilation

### ISSUE THREAD

**Issue by Ahleroy** [isl-org/Open3D#3464]
I just compiled Open3D on Ubuntu 18.04. 
When using CMake, OpenMP is correctly detected. Please find below the output of "cmake .." in the "build" folder : 

```
-- THIRD_PARTY_DOWNLOAD_DIR is not specified, will download directly.
-- No CMAKE_BUILD_TYPE specified, default to RelWithDebInfo
CMake Deprecation Warning at CMakeLists.txt:96 (cmake_policy):
  The OLD behavior for policy CMP0072 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Open3D 0.12.0+313315d97
-- Using python from PYTHON_EXECUTABLE variable: /home/emmab/miniconda3/bin/python
-- (PYTHON_EXECUTABLE matches python from PATH)
-- Building library Assimp from source
-- Building with OpenMP
-- Building library 3rdparty_eigen3 from source
-- Building library 3rdparty_flann from source
-- Building library 3rdparty_nanoflann from source
-- Building library 3rdparty_glew from source
-- Building library 3rdparty_glfw3 from source
-- Could NOT find Vulkan (missing: VULKAN_LIBRARY VULKAN_INCLUDE_DIR) 
-- Using X11 for window creation
-- Building third-party library JPEG from source
-- NASM assembler not found - libjpeg-turbo performance may suffer
-- libturbojpeg: WITH_CRT_DLL=OFF
-- Building library 3rdparty_jsoncpp from source
-- Building library 3rdparty_lzf from source
-- Building library 3rdparty_tritriintersect from source
-- Building third-party library zlib from source
-- Building third-party library libpng from source
-- Building library 3rdparty_rply from source
-- Building library 3rdparty_tinyfiledialogs from source
-- Building library 3rdparty_tinygltf from source
-- Building library 3rdparty_tinyobjloader from source
-- Building library 3rdparty_qhull_r from source
-- Building library 3rdparty_qhullcpp from source
-- Building library 3rdparty_fmt from source
-- pybind11 v2.5.0
-- Building library 3rdparty_poisson from source
-- Building library 3rdparty_imgui from source
-- Using prebuilt third-party library Filament
-- GLIBC version 2.27-3 found: Downloading Google Filament binary.
-- Filament is located at /home/alix/Documents/open3d/Open3D/build/filament/src/ext_filament
-- CLANG_LIBDIR: /usr/lib/x86_64-linux-gnu
-- CPP_LIBRARY: /usr/lib/x86_64-linux-gnu/libc++.so
-- CPPABI_LIBRARY: /usr/lib/x86_64-linux-gnu/libc++abi.so
-- Using remote url(s): https://anaconda.org/intel/mkl-static/2020.1/download/linux-64/mkl-static-2020.1-intel_217.tar.bz2
-- Using remote url(s): https://github.com/intel-isl/Open3D/releases/download/v0.10.0/linux-merged-mkl-static-2020.1-intel_217.zip
-- Using MKL to support BLAS and LAPACK functionalities.
-- STATIC_MKL_INCLUDE_DIR: /home/alix/Documents/open3d/Open3D/build/mkl_install/include/
-- STATIC_MKL_LIB_DIR: /home/alix/Documents/open3d/Open3D/build/mkl_install/lib
-- STATIC_MKL_LIBRARIES: mkl_merged;tbb_static
-- Building third-party library faiss from source
-- FAISS_INCLUDE_DIR: /home/alix/Documents/open3d/Open3D/build/mkl_install/include/
-- FAISS_LIB_DIR: /home/alix/Documents/open3d/Open3D/build/mkl_install/lib
-- BUILD_JUPYTER_EXTENSION is set to OFF
-- Using Python version: 3.8
-- Using Python version major: 3
-- PYPI_PACKAGE_NAME: open3d
-- ========================================================================
-- Open3D 0.12.0 Configuration Summary
-- ========================================================================
-- Enabled Features:
--   OpenMP .................................. ON
--   Headless Rendering ...................... OFF
--   Azure Kinect Support .................... OFF
--   Intel RealSense Support ................. OFF
--   CUDA Support ............................ OFF
--   Build GUI ............................... ON
--   Build Shared Library .................... OFF
--   Build Unit Tests ........................ OFF
--   Build Examples .......................... 
--   Build Python Module ..................... ON
--   - with Jupyter Notebook Support ......... OFF
--   Build TensorFlow Ops .................... OFF
--   Build PyTorch Ops ....................... OFF
--   Build Benchmarks ........................ OFF
--   Bundle Open3D-ML ........................ OFF
--   Build RPC interface ..................... OFF
--   Force GLIBCXX_USE_CXX11_ABI= ............ 0
-- ========================================================================
-- Third-Party Dependencies:
--   EIGEN3 .................................. yes (build from source)
--   FAISS ................................... yes (build from source)
--   FILAMENT ................................ yes (build from source)
--   FLANN ................................... yes (build from source)
--   FMT ..................................... yes (build from source)
--   GLEW .................................... yes (build from source)
--   GLFW .................................... yes (build from source)
--   GOOGLETEST .............................. no
--   IMGUI ................................... yes (build from source)
--   JPEG .................................... yes (build from source)
--   JSONCPP ................................. yes (build from source)
--   LIBLZF .................................. yes (build from source)
--   OPENGL .................................. yes (build from source)
--   PNG ..................................... yes (build from source)
--   PYBIND11 ................................ yes (build from source)
--   QHULL ................................... yes (build from source)
--   LIBREALSENSE ............................ no
--   TINYFILEDIALOGS ......................... yes (build from source)
--   TINYGLTF ................................ yes (build from source)
--   TINYOBJLOADER ........................... yes (build from source)
-- ========================================================================
-- Configuring done
-- Generating done
-- Build files have been written to: /home/alix/Documents/open3d/Open3D/build

````

I successfully compiled Open3D and I was able to install it to my python virtual environment using "make install-pip-package".

Python found the Open3D module with the following version : 0.12.0+313315d97

I have been able to run my python scripts, however they do not seem to take advantage of OpenMP.
So far, I tried the following functions : 
- registration_fast_based_on_feature_matching
- registration_ransac_based_on_feature_matching

Only one of my 64 CPU cores is used during the registration.

EDIT:

Poisson reconstruction seems to take advantage of all the CPU cores. Does this mean the registration methods are not compatible with OpenMP ?

**Comment by ssheorey** [comment]
Hi @Ahleroy all functions do not use OpenMP for parallel processing. The newer Tensor based API extensively uses OpenMP / TBB to parallelize tasks. Please try out the Tensor based functions from here:

http://www.open3d.org/docs/latest/python_api/open3d.t.pipelines.registration.html

We are still adding to the functionality here,  so everything is not available yet.

If you see any low hanging fruit for parallelization, PRs are welcome.

**Comment by Ahleroy** [comment]
Looks like this is only available for the master branch and not the 0.12.0 version right ?

**Comment by ssheorey** [comment]
Yes that's correct. We plan to release this in v0.13 next week. Please try it out in `master` (or with a development Python  wheel) and provide feedback.

http://www.open3d.org/docs/latest/getting_started.html#development-version-pip
